server/src/test/java/org/opensearch/cluster/routing/allocation/TieringAllocationBaseTestCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.routing.allocation;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+
+import static org.opensearch.index.IndexModule.INDEX_STORE_LOCALITY_SETTING;
+import static org.opensearch.index.IndexModule.INDEX_TIERING_STATE;
+
+@SuppressForbidden(reason = "feature flag overrides")
+public abstract class TieringAllocationBaseTestCase extends RemoteShardsBalancerBaseTestCase {
+
+    @BeforeClass
+    public static void setup() {
+        System.setProperty(FeatureFlags.TIERED_REMOTE_INDEX, "true");
+    }
+
+    @AfterClass
+    public static void teardown() {
+        System.setProperty(FeatureFlags.TIERED_REMOTE_INDEX, "false");
+    }
```

Comment:
Can we use `FeatureFlagSetter#set` here instead? And then cleanup should be handled by `OpenSearchTestCase`:
https://github.com/opensearch-project/OpenSearch/blob/1084ba9b167af8342be0acea075c83fa1673dcb1/test/framework/src/main/java/org/opensearch/test/OpenSearchTestCase.java#L242-L247